pkg/controller/machine/machine_suite_test.go

```diff
@@ -22,33 +22,33 @@ import (
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/test"
 
 	"sigs.k8s.io/cluster-api/pkg/apis"
-	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1/testutil"
 	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 	"sigs.k8s.io/cluster-api/pkg/openapi"
 )
 
-type fakeActuator struct{}
-
-func (fakeActuator) Create(*clusterv1.Cluster, *clusterv1.Machine) error           { return nil }
-func (fakeActuator) Delete(*clusterv1.Machine) error                               { return nil }
-func (fakeActuator) Update(c *clusterv1.Cluster, machine *clusterv1.Machine) error { return nil }
-func (fakeActuator) Exists(*clusterv1.Machine) (bool, error)                       { return false, nil }
-
 func TestMachine(t *testing.T) {
 	testenv := test.NewTestEnvironment()
 	config := testenv.Start(apis.GetAllApiBuilders(), openapi.GetOpenAPIDefinitions)
 	cs := clientset.NewForConfigOrDie(config)
 
 	shutdown := make(chan struct{})
 	si := sharedinformers.NewSharedInformers(config, shutdown)
-	controller := NewMachineController(config, si, fakeActuator{})
-	controller.Run(shutdown)
-
-	t.Run("machineControllerReconcile", func(t *testing.T) {
-		machineControllerReconcile(t, cs, controller)
-	})
-
+	actuator := &TestActuator{}
+	controller := NewMachineController(config, si, actuator)
+	controller.RunAsync(shutdown)
+
+	// Create a cluster object.
+	cluster := testutil.GetVanillaCluster()
+	cluster.Name = "cluster-1"
+	if _, err := cs.ClusterV1alpha1().Clusters("default").Create(&cluster); err != nil {
+		t.Fatal(err)
+	}
+
+	// It's intentionally to run those test in sequence to simplify the test setup.
+	machineControllerReconcile(t, cs, controller)
```

Comment:
I don't understand why you removed the calls to `t.Run`. They provide important separation of sub-tests both in testing framework (a Fatal in the first test can't stop the second from running), and in reporting.

Reply:
I'm having difficult to set up two test environment to run in parallel. I got following error. Do you have hint? 

 panic: the group "cluster.k8s.io" has already been announced.

goroutine 88 [running]:
sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/apimachinery/announced.(*GroupMetaFactory).Announce(0xc420d8ef90, 0xc420335e90, 0xc420d8ef90)
	/usr/local/google/home/fmin/.gvm/pkgsets/go1.9.4/global/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go:85 +0x6c